nemo/collections/asr/data_layer.py

```diff
@@ -39,6 +40,41 @@
 logging = nemo.logging
 
 
+def _process_augmentations(augmenter) -> AudioAugmentor:
+    if isinstance(augmenter, AudioAugmentor):
+        return augmenter
+
+    if not type(augmenter) == dict:
+        raise ValueError("Cannot parse augmenter. Must be a dict or an AudioAugmentor object ")
+
+    augmenter = copy.deepcopy(augmenter)
+
+    augmentations = []
+    for augment_name, augment_kwargs in augmenter.items():
+        prob = augment_kwargs.get('prob', None)
+
+        if prob is None:
+            logging.warning(
+                f'Augmentation "{augment_name}" will not be applied as '
+                f'keyword argument "prob" was not defined for this augmentation.'
+            )
+
+        else:
+            _ = augment_kwargs.pop('prob')
+
+            if prob < 0.0 or prob > 1.0:
+                raise ValueError("`prob` must be a float value between 0 and 1.")
+
+            try:
+                augmentation = perturbation_types[augment_name](**augment_kwargs)
+                augmentations.append([prob, augmentation])
+            except KeyError:
```

Comment:
Don't except this, just raise KeyError

Reply:
Fail fast behavior? Would it be better to catch it, and then reraise with a more informative message ?

Reply:
Sure you can catch it can reraise, but elevate from logging.warning to an error

Reply:
Sounds good.

Reply:
Raised KeyError here.